### PR TITLE
add get_step_failure_events to RunFailureSensorContext

### PR DESCRIPTION
**Pull request by sryza**
## Summary & Motivation

Helps with https://github.com/dagster-io/dagster/issues/10912

### CODE REVIEW

**Comment by sryza** [comment]
@jamiedemaria or @yuhan - mind taking a look at this now that the dust has settled on the latest release?

File: python_modules/dagster/dagster/_core/definitions/run_status_sensor_definition.py [hunk 1]
```diff
@@ -253,10 +253,33 @@ class RunFailureSensorContext(RunStatusSensorContext):
         failure_event (DagsterEvent): the pipeline failure event.
```
Comment (jamiedemaria) [inline python_modules/dagster/dagster/_core/definitions/run_status_sensor_definition.py:253]:
```suggestion
        failure_event (DagsterEvent): the run failure event.
```
nit - make consistent with the property doc string

Comment (jamiedemaria) [reply python_modules/dagster/dagster/_core/definitions/run_status_sensor_definition.py:253]: can't make a comment on the above line, but could also change 
```
dagster_run (DagsterRun): the failed pipeline run.
```
to 
```
dagster_run (DagsterRun): the failed run.
```

**Review by jamiedemaria — APPROVED** [review]
looks good to me! one very small comment. i really like that the doc block for failure_event tells you what function to use to get more information!

**Pull request merged**